Updated explanation when maximum timeout will be used (#2021)
buildkite/docs: pages/pipelines/command_step.md

Specific timeouts take precedence over more general ones — a step level timeout takes precedence over a pipeline timeout, which in turn takes precedence over an organization default.

> Maximum timeouts are applied to command steps in the following situations:

Reviewer: Style: You need to include a blank line before a list for it to render correctly. At the moment, it looks like:

<img width="741" alt="Screenshot 2023-04-20 at 8 20 54 am" src="https://user-images.githubusercontent.com/119824349/233212370-4ab8f0f5-a778-4278-9685-4ba3580592ca.png">

```suggestion
Maximum timeouts are applied to command steps in the following situations:

```

- No timeout attribute is set on the step.
- No default timeout is set in the pipeline settings.

> - When the timeout set is greater than the maximum timeout

Reviewer: Typo: Missing period.

```suggestion
- When the timeout set is greater than the maximum timeout.
```

Specific timeouts take precedence over more general ones — a step level timeout takes precedence over a pipeline timeout, which in turn takes precedence over an organization default.

> Maximum timeouts are applied to any command step without a timeout, default timeout not set in pipeline settings or with a timeout greater than the maximum timeout.

Reviewer: Style: Use a serial comma.

```suggestion
Maximum timeouts are applied to any command step without a timeout, default timeout not set in pipeline settings, or with a timeout greater than the maximum timeout.
```

Reviewer: Actually, I think the issue is how hard the sentence is to parse. How about splitting it into a bulleted list? Thinking:

```suggestion
Maximum timeouts are applied to command steps in the following situations:

- No timeout attribute is set on the step.
- No default timeout is set in the pipeline settings.
- When the timeout set is greater than the maximum timeout.
```

Author: @mbelton-buildkite I like this bullet list approach as it is easy to read and understand. I updated the PR with this approach
